packages/titus-backend-typescript/src/config/index.ts

```diff
@@ -0,0 +1,139 @@
+    AUTH0_CONNECTION: Type.Optional(Type.String()),
+    COGNITO_REGION: Type.Optional(Type.String()),
+    COGNITO_USER_POOL_ID: Type.Optional(Type.String()),
+    JWT_SECRET: Type.String({ default: '3000' }),
+    AD_TENANT: Type.Optional(Type.String()),
+    AD_APP_ID: Type.Optional(Type.String()),
+    AD_SECRET: Type.Optional(Type.String()),
+    SECRETS_STRATEGY: Type.Optional(Type.String()),
+    SECRETS_PG_PASS: Type.String(),
+    HEALTHCHECK_URL: Type.String({
+      default: '/healthcheck'
+    }),
+    HEALTHCHECK_MAX_HEAP_USER: Type.Number({
+      default: 768 * 1024 * 1024
+    }), // arbitrary, 768 MB of RA,
+    HEALTHCHECK_MAX_RSS: Type.Number({
+      default: 1024 * 1024 * 1024
+    }), // arbitrary, 1 GB of RA,
+    HEALTHCHECK_MAX_EVENT_LOOP_UTILIZATION: Type.Number({
+      default: 0.98
+    })
+  })
+)
+
+const config = envSchema<Static<typeof envJsonSchema>>({
+  dotenv: true,
+  schema: envJsonSchema
+})
+
+const routeResponseSchemaOpts = Type.Strict(
+  Type.Object({
+    version: Type.String(),
+    serverTimestamp: Type.String(),
+    db: Type.String(),
+
+    memoryUsage: Type.Object({
+      eventLoopDelay: Type.String(),
+      rssBytes: Type.String(),
+      heapUsed: Type.String()
+    })
+  })
+).properties
+
+const isProduction = /^\s$production\s*$/i.test(config.NODE_ENV)
+
+// Global configuration, from env variables
+export const configOptions = {
+  isProduction,
+  server: {
+    host: config.API_HOST,
+    port: config.API_PORT
+  },
+  fastify: {
+    logger: true
+  },
+  enableAdmin: config.ENABLE_ADMIN,
+  pgPlugin: {
+    host: config.PG_HOST,
+    port: config.PG_PORT,
+    database: config.PG_DB,
+    user: config.PG_USER,
+    poolSize: 10,
+    idleTimeoutMillis: 30000
+  } as PostgresPluginOptions & PoolConfig,
+  underPressure: {
+    maxHeapUsedBytes: config.HEALTHCHECK_MAX_HEAP_USER,
+    maxRssBytes: config.HEALTHCHECK_MAX_RSS,
+    maxEventLoopUtilization: config.HEALTHCHECK_MAX_EVENT_LOOP_UTILIZATION,
+    exposeStatusRoute: {
+      url: config.HEALTHCHECK_URL,
+      routeResponseSchemaOpts
+    }
+  } as underPressurePlugin.UnderPressureOptions,
+  cors: { origin: !!config.CORS_ORIGIN, credentials: true },
+  auth: {
+    provider: config.AUTH_PROVIDER || 'auth0',
+    azureAD: {
+      appID: config.AD_APP_ID,
+      secret: config.AD_SECRET,
+      tenant: config.AD_TENANT
+    },
+    auth0: {
+      domain: config.AUTH0_DOMAIN,
+      clientId: config.AUTH0_CLIENT_ID,
+      clientSecret: config.AUTH0_CLIENT_SECRET,
+      audience: config.AUTH0_AUDIENCE,
+      grantType: config.AUTH0_GRANT_TYPE,
+      connection: config.AUTH0_CONNECTION
+    },
+    cognito: {
+      region: config.COGNITO_REGION,
+      userPoolId: config.COGNITO_USER_POOL_ID
+    }
+  },
+  jwt: {
+    secret: config.JWT_SECRET
+  },
+  secretsManager: {
+    strategy: config.SECRETS_STRATEGY,
+    secrets: {
+      dbPassword: config.SECRETS_PG_PASS
+    },
+    clientOptions: null as any
```

Comment:
can this stay as `null`?
```suggestion
    clientOptions: null
```